oci/oci.go

```diff
@@ -514,20 +514,34 @@ func (r *Runtime) StopContainer(c *Container, timeout int64) error {
 		timeout = 10
 	}
 	done := make(chan struct{})
+	// we could potentially re-use "done" channel to exit the loop on timeout
+	// but we use another channel "chControl" so that we won't never incur in the
+	// case the "done" channel is closed in the "default" select case and we also
+	// reach the timeout in the select below. If that happens we could raise
+	// a panic closing a closed channel so better be safe and use another new
+	// channel just to control the loop.
+	chControl := make(chan struct{})
 	go func() {
 		for {
-			// Check if the process is still around
-			err := unix.Kill(c.state.Pid, 0)
-			if err == syscall.ESRCH {
-				close(done)
-				break
+			select {
+			case <-chControl:
+				return
+			default:
+				// Check if the process is still around
+				err := unix.Kill(c.state.Pid, 0)
+				if err == syscall.ESRCH {
+					close(done)
+					return
+				}
+				time.Sleep(100 * time.Millisecond)
 			}
```

Comment:
No, when timeout is hit, at next iteration we hit https://github.com/kubernetes-incubator/cri-o/pull/602/files#diff-87558c1a90a59158e2c18a316e0c6693R527 which is closed and goes to return and exit the goroutine